### ISSUE TITLE
Parenthesis around table and alias in join in PostgreSQL is not permitted, as in MySQL

### ISSUE THREAD

**Issue by krasandr** [darold/ora2pg#606]
MySQL permits parenthesis around table in join, for example original code:
```
| f_ivfprocprice | STRICT_TRANS_TABLES,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION | CREATE DEFINER=`root`@`%` FUNCTION `f_ivfprocprice`(ArtID INT(11)) RETURNS decimal(12,2)
BEGIN

  RETURN (ifnull((SELECT sum(fin.JAmount)
                  FROM
                    jorders fin
                  INNER JOIN (services s)
                  ON (s.ServiceID = fin.ServiceID)
                  WHERE
                    fin.IvfArtID = ArtID
                    AND fin.IsInclusive = 0
                    AND s.ServiceCodeMkb9 IN ('C_1EHO', 'C_2HOR', 'C_3PUNC', 'C_4ICSI', 'C_5IVF', 'C_6TRAN', 'C_7CRYO')
  ), 0));
END | utf8                 | utf8_general_ci      | latin1_swedish_ci  |
```
PostgreSQL don't permit parenthesis (tested with 10.1 version), so converted code don't work:
```
CREATE OR REPLACE FUNCTION f_ivfprocprice (ArtID bigint) RETURNS decimal AS $body$
BEGIN

  RETURN (COALESCE((SELECT sum(fin.JAmount)
                  FROM
                    jorders fin
                  INNER JOIN(services s)
                  ON (s.ServiceID = fin.ServiceID)
                  WHERE
                    fin.IvfArtID = ArtID
                    AND fin.IsInclusive = 0
                    AND s.ServiceCodeMkb9 IN ('C_1EHO', 'C_2HOR', 'C_3PUNC', 'C_4ICSI', 'C_5IVF', 'C_6TRAN', 'C_7CRYO')
  ), 0));
END;
$body$
LANGUAGE PLPGSQL
SECURITY DEFINER
;
-- REVOKE ALL ON FUNCTION f_ivfprocprice (ArtID bigint) FROM PUBLIC;
```
If I remove parenthesis around INNER JOIN(services s) to become INNER JOIN services s
everything works. This is not a big problem, but is something related to syntax.

**Comment by krasandr** [comment]
Also, seems MySQL permit mixed style JOIN (with comma and with ON). PostgreSQL don't permit them (I think, that ANSI standard also don't permit them), but only to report this issue (I think, there is no way to fix it in Ora2Pg, proper is to rewrite JOIN as ANSI only):

| f_checkisinclusiveservice | STRICT_TRANS_TABLES,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION | CREATE DEFINER=`root`@`%` FUNCTION `f_checkisinclusiveservice`(p_depPatID int, p_SerID int) RETURNS int(11)
BEGIN

  RETURN (SELECT
      IF(s.IsInclusive = 0, 0, IF(dp.DepartmentPationID IS NULL, 1, 0))
    FROM services s
      LEFT JOIN (clinic_paths_checks ch,
                 department_patients dp)
                   ON (ch.Mkb9CharsCode = s.ServiceCodeMkb9
                   AND ch.ClinicPathID = dp.OutClinicPathID
                   AND dp.DepartmentPationID = p_depPatID)
    WHERE s.ServiceID = p_SerID);
END | utf8                 | utf8_general_ci      | latin1_swedish_ci  |

**Comment by darold** [comment]
First case is solved with commit 7e10905 but the complex form just above is not easy to translate and will need lot of work. I don't think I will do this job because unlink (+) in Oracle I think there is not lot of code using it so I don't think the effort should be done.

**Comment by krasandr** [comment]
I think the same, for example in my work for Customs, I work about 15 years with IBM Informix SQL databases - they are very closely related to PostgreSQL (because Michael Stonebraker was head of development of both databases for years), and Informix simply refuses mixing of ANSI and Informix style of joins in one query (it looks like Oracle, but not with a "+" sign, Informix syntax is outer(table_name) with ",", for exampe select ... from x1,x2,outer(x3,x4,outer(x5)) where ... (there is no ON, like in Oracle, when you use (+).
So I don't know why MySQL permits mixing of both join styles and also, it is right solution such code simply to be rewritten in right manner, not converted.
